### PR TITLE
added check for multipicklist other class as not being binary in metadata

### PR DESCRIPTION
**Pull request by leahmcguire**
**Related issues**
creation of spark attribute groups was mislabeling multipicklist other column as binary. this made tree models fail

**Describe the proposed solution**
multipicklist other is now a numeric not binary

### CODE REVIEW

File: features/src/main/scala/com/salesforce/op/utils/spark/OpVectorColumnMetadata.scala [hunk 1]
```diff
@@ -155,6 +161,7 @@ object OpVectorColumnMetadata {
   val DescriptorValueKey = "descriptor_value"
   val IndicesKey = "indices"
   val NullString = "NullIndicatorValue"
+  val OtherString = "OTHER"
```
[the remark this answers is not in the document]
Comment (leahmcguire) [reply features/src/main/scala/com/salesforce/op/utils/spark/OpVectorColumnMetadata.scala]: I changed the TransmogrifierDefaults to point to this

**Comment by michaelweilsalesforce** [comment]
@leahmcguire should we add in `OpRandomForestTest` a unit test on a data that contains almost all possible feature transformation we have?
I can help contribute to it...

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala [hunk 2]
```diff
@@ -42,8 +43,11 @@ trait AttributeAsserts {
    * @param schema
    * @param expectedNominal Expected array of booleans. True if the field is nominal, false if not.
    */
-  final def assertNominal(schema: StructField, expectedNominal: Array[Boolean]): Assertion = {
+  final def assertNominal(schema: StructField, expectedNominal: Array[Boolean], output: Array[OPVector]): Assertion = {
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala]: please update docs for the `assertNominal` function

File: core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala [hunk 3]
```diff
@@ -42,8 +43,11 @@ trait AttributeAsserts {
...
     val attributes = AttributeGroup.fromStructField(schema).attributes
+    output.foreach{
+      x => x.value.toArray.zip(expectedNominal).foreach{ case (v , n) => if (n) (v == 0.0 || v == 1.0) shouldBe true }
```
Comment (tovbinm) [inline core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala]: should we also add a check when `n == false`? is so, then better:
```scala
for {
    x <- output
    (value, nominal) <- x.value.toArray.zip(expectedNominal)
} if (nominal) (v == 0.0 || v == 1.0) shouldBe true else ???
```

Comment (leahmcguire) [reply core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala]: there are no constraints on if n is false

Comment (tovbinm) [reply core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala]: so it can be NaN, -/+Inf as well?

**Pull request merged**